exonum-java-binding/core/src/main/java/com/exonum/binding/core/service/AbstractService.java

```diff
@@ -58,11 +55,6 @@ protected final ServiceInstanceSpec getInstanceSpec() {
     return instanceSpec;
   }
 
-  @Override
-  public List<HashCode> getStateHashes(Snapshot snapshot) {
-    return createDataSchema(snapshot).getStateHashes();
-  }
-
   /**
    * Creates a data schema of this service.
```

Comment:
(May be offloaded for discussion to a sep. issue — please add)
Shall this method remain? It's not just enough to override it, the return type must
also be adjusted (e.g., `@Override QaSchema createDataSchema(View) { }` in QaService)

The (small) value it still provides is that it explicitly connects a service to a schema.

Reply:
ECR-4029